docs/how-to/hip_porting_driver_api.rst

```diff
@@ -0,0 +1,480 @@
+
+        .. code-block:: cpp
+
+            CUmodule module;
+            void *imagePtr = ...; // Somehow populate data pointer with code object
+
+            const int numOptions = 1;
+            CUJit_option options[numOptions];
+            void *optionValues[numOptions];
+
+            options[0] = CU_JIT_MAX_REGISTERS;
+            unsigned maxRegs = 15;
+            optionValues[0] = (void *)(&maxRegs);
+
+            cuModuleLoadDataEx(module, imagePtr, numOptions, options, optionValues);
+
+            CUfunction k;
+            cuModuleGetFunction(&k, module, "myKernel");
+
+The sample below shows how to use ``hipModuleGetFunction``.
+
+.. code-block:: cpp
+
+    #include <hip_runtime.h>
+    #include <hip_runtime_api.h>
+
+    #include <fstream>
+    #include <iostream>
+    #include <vector>
+
+    #define LEN 64
+    #define SIZE LEN << 2
+
+    #ifdef __HIP_PLATFORM_AMD__
+    #define fileName "vcpy_isa.co"
+    #endif
+
+    #ifdef __HIP_PLATFORM_NVIDIA__
+    #define fileName "vcpy_isa.ptx"
+    #endif
+
+    #define kernel_name "hello_world"
+
+    int main() {
+        float *A, *B;
+        hipDeviceptr_t Ad, Bd;
+        A = new float[LEN];
+        B = new float[LEN];
+
+        for (uint32_t i = 0; i < LEN; i++) {
+            A[i] = i * 1.0f;
+            B[i] = 0.0f;
+            std::cout << A[i] << " " << B[i] << std::endl;
+        }
+
+        #ifdef __HIP_PLATFORM_NVIDIA__
+        hipInit(0);
+        hipDevice_t device;
+        hipCtx_t context;
+        hipDeviceGet(&device, 0);
+        hipCtxCreate(&context, 0, device);
+        #endif
+
+        hipMalloc((void **)&Ad, SIZE);
+        hipMalloc((void **)&Bd, SIZE);
+
+        hipMemcpyHtoD(Ad, A, SIZE);
+        hipMemcpyHtoD(Bd, B, SIZE);
+        hipModule_t Module;
+        hipFunction_t Function;
+        hipModuleLoad(&Module, fileName);
+        hipModuleGetFunction(&Function, Module, kernel_name);
+
+        std::vector<void *> argBuffer(2);
+        memcpy(&argBuffer[0], &Ad, sizeof(void *));
+        memcpy(&argBuffer[1], &Bd, sizeof(void *));
+
+        size_t size = argBuffer.size() * sizeof(void *);
+
+        void *config[] = {HIP_LAUNCH_PARAM_BUFFER_POINTER, &argBuffer[0],
+                            HIP_LAUNCH_PARAM_BUFFER_SIZE, &size, HIP_LAUNCH_PARAM_END};
+
+        hipModuleLaunchKernel(Function, 1, 1, 1, LEN, 1, 1, 0, 0, NULL,
+                                (void **)&config);
+
+        hipMemcpyDtoH(B, Bd, SIZE);
+        for (uint32_t i = 0; i < LEN; i++) {
+            std::cout << A[i] << " - " << B[i] << std::endl;
+        }
+
+        #ifdef __HIP_PLATFORM_NVIDIA__
+        hipCtxDetach(context);
+        #endif
+
+        return 0;
+    }
+
+HIP module and texture Driver API
+=================================
+
+HIP supports texture driver APIs. However, texture references should be declared within the host scope. The following code demonstrates the use of texture references for the ``__HIP_PLATFORM_AMD__`` platform.
```

Comment:
Is it mandatory to declare the references in the host scope? If so, use "must" instead of "should".